Refactor `examples/mask.py`

pycoinfu opened the pull request:
I made `pygame.examples.mask` use `pygame.Vector2`s, follow Python conventions (currently it's using camelCase for object naming) and simplified the code by removing some of the functions that weren't necessary (`Sprite.setPos`, `Sprite.setVelocity`, etc.)
After asking some people on Discord, I also removed the Python/C performance comparison, because it was calling the exact same method, just the python one was wrapped in another function

illume commented:
thanks-very-cool

2x combo points if you could run the python black code formatter on it? 
8x combo points if pylint passes too :)

pycoinfu commented:
> thanks-very-cool
> 
> 2x combo points if you could run the python black code formatter on it? 8x combo points if pylint passes too :)

Thanks for checking out my PR! 😄 
I added missing docstrings and changed the variable names that pylint was complaining about
The code was already `black`-ified, is there some setting I should use?

novialriptide commented on examples/mask.py:
```diff
@@ -204,8 +195,8 @@ def main(*args):
         main_dir = os.path.split(os.path.abspath(__file__))[0]
-        imagename = os.path.join(main_dir, "data", "alien1.png")
-        main(imagename)
+        image_name = os.path.join(main_dir, "data", "alien1.png")
```
Should this be condensed into one line?

novialriptide commented on examples/mask.py:
```diff
@@ -132,70 +128,65 @@ def main(*args):
-            return
+        for event in pg.event.get():
+            if event.type in (pg.QUIT, pg.KEYDOWN):
+                return
```
Other pygame examples use `break` instead of `return`, this should be changed here to stay consistent.

pycoinfu replied:
That would just break out of the event loop, and using `pygame.event.wait` in this case isn't possible because that would cause the sprites to stop moving

illume reviewed (approved):
🎉 🎈

ankith26 commented on examples/mask.py:
```diff
@@ -36,92 +37,87 @@
+    Moving Sprite demonstrating pixel-perfect collisions between pg.mask.Mask objects
+    """
+
+    def __init__(self, pos, vel, surface: pg.Surface, mask: pg.mask.Mask = None):
```
I think this is inconsistent typehinting because some args are typed and some are not, it should be all or none IMHO (I'm personally leaning on the latter for now because most of our other examples don't have type hints, but I'll let you take the call)

If you do decide to go with the typehinting, there is a small issue here: The `mask` argument should be typed like `Optional[pg.mask.Mask]`

pycoinfu replied:
Thanks for checking out my PR! 🥳 
Decided to remove the typehints

ankith26 reviewed (approved):
LGTM! Thanks for the PR disa 🚀 🍕

novialriptide commented:
🥳

pycoinfu commented:
Thanks everyone! 💜
